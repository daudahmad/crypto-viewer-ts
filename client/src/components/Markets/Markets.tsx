import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import styled from "react-emotion";

import TickerList from "../TickerList";
import { PairsState, StoreState } from "../../types";
import {
  TickerActions,
  requestPairs,
  selectTicker
} from "../../actions";
import Loading from "../Loading";
import { select } from "redux-saga/effects";
import MarketSymbol from "../MarketSymbol";

interface Props extends PairsState {
  fetchPairs: () => void;
  selectTicker: (tickerSymbol: string) => void;
}

export class Markets extends Component<Props, {}> {
  constructor(props: any) {
    super(props);
  }

  componentDidMount() {
    this.props.fetchPairs();
  }

  handleSelectMarketClick = (tickerSymbol: string) => {
    this.props.selectTicker(tickerSymbol);
  };

  render() {
    const { isFetching, items} = this.props;
    return (
      <Fragment>
        {isFetching && <Loading />}
        {!isFetching && <TickerList pairs={items} onSelectMarketClick={this.handleSelectMarketClick} />}
      </Fragment>
    );
  }
}

const mapDispatchToProps = (dispatch: Dispatch<TickerActions>) => ({
  fetchPairs() {
    dispatch(requestPairs());
  },
  selectTicker(tickerSymbol: string) {
    dispatch(selectTicker(tickerSymbol));
  }
});

const mapStateToProps = ({
  pairs: { items, isFetching }
}: {
  pairs: PairsState;
}) => ({
  items,
  isFetching
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Markets);
