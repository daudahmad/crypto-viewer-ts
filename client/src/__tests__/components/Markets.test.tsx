import React from "react";
import { create } from "react-test-renderer";
import { Markets } from "../../components/Markets/Markets";

describe("snapshot", () => {
  const fetchPairs = jest.fn();
  const selectTicker = jest.fn();
  const c = create(
    <Markets
      fetchPairs={fetchPairs}
      selectTicker={selectTicker}
      items={[]}
      isFetching={false}
    />
  );
  console.log(c.toJSON());
  expect(c.toJSON()).toMatchSnapshot();
});
