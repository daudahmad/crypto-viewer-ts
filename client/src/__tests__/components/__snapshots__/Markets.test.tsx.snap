// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <h4>
    Ticker Details
  </h4>,
  <ul
    className="MuiList-root-1 MuiList-padding-2"
  />,
]
`;
